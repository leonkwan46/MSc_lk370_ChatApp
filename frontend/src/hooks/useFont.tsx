import { useFonts } from 'expo-font'

const useFont = (font: string) => {
    const [fontLoaded, fontError] = useFonts({
        [font]: require('../assets/fonts/Lemon-Regular.ttf')
    })
    return { fontLoaded, fontError }
}

export default useFont