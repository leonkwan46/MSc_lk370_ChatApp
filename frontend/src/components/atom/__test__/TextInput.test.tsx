import React from 'react'
import renderer from 'react-test-renderer'
import CustomTextInput from '../TextInput'
import { TextInput } from 'react-native'
import useFont from '@/src/hooks/useFont'

jest.mock('@/src/hooks/useFont', () => ({
    __esModule: true,
    default: jest.fn().mockReturnValue({ fontLoaded: true })
}))
  

jest.mock('@/src/helpers/styleHelper', () => ({
  getTextInputStyle: jest.fn(() => ({ borderWidth: 1 }))
}))

describe('CustomTextInput Component', () => {
  it('renders correctly with required props', () => {
    const tree = renderer
      .create(
        <CustomTextInput
          value="Test Value"
          size="medium"
          onChangeText={() => {}}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with placeholder and secureTextEntry', () => {
    const tree = renderer
      .create(
        <CustomTextInput
          value=""
          size="medium"
          placeholder="Enter text"
          secureTextEntry
          onChangeText={() => {}}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with error state and custom styles', () => {
    const extraStyles = { borderColor: 'red', borderWidth: 2 }
    const tree = renderer
      .create(
        <CustomTextInput
          value="Error State"
          size="medium"
          hasError
          extrasStyle={extraStyles}
          onChangeText={() => {}}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when font is not loaded', () => {
    (useFont as jest.Mock).mockReturnValue({ fontLoaded: false })

    const tree = renderer
      .create(
        <CustomTextInput
          value="No Font"
          size="medium"
          onChangeText={() => {}}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('calls onChangeText when text changes', () => {
    const onChangeTextMock = jest.fn()
    const testRenderer = renderer.create(
      <CustomTextInput
        value=""
        size="medium"
        onChangeText={onChangeTextMock}
      />
    )

    const textInput = testRenderer.root.findByType(TextInput)
    textInput.props.onChangeText('New Text')

    expect(onChangeTextMock).toHaveBeenCalledWith('New Text')
  })

  it('renders with disabled state', () => {
    const tree = renderer
      .create(
        <CustomTextInput
          value="Disabled"
          size="medium"
          onChangeText={() => {}}
          disabled
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
