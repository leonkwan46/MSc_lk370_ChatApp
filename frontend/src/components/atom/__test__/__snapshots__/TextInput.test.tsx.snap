// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextInput Component renders correctly when font is not loaded 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    placeholderTextColor="#aaa"
    style={null}
    value="No Font"
  />
</View>
`;

exports[`CustomTextInput Component renders correctly with required props 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    placeholderTextColor="#aaa"
    style={
      {
        "borderWidth": 1,
      }
    }
    value="Test Value"
  />
</View>
`;

exports[`CustomTextInput Component renders with disabled state 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    placeholderTextColor="#aaa"
    style={null}
    value="Disabled"
  />
</View>
`;

exports[`CustomTextInput Component renders with error state and custom styles 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    placeholderTextColor="#aaa"
    style={
      {
        "borderColor": "red",
        "borderWidth": 2,
      }
    }
    value="Error State"
  />
</View>
`;

exports[`CustomTextInput Component renders with placeholder and secureTextEntry 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter text"
    placeholderTextColor="#aaa"
    secureTextEntry={true}
    style={
      {
        "borderWidth": 1,
      }
    }
    value=""
  />
</View>
`;
